docs/rfcs/20220209_consumer_offsets.md

```diff
@@ -0,0 +1,190 @@
+to interact with the `__consumer_offsets` topic. Additionally, we may want to add 
+decoding capabilities so that `rpk` will be able to display human/machine 
+readable content from the `__consumer_offsets` topic.
+
+### Admin API
+
+There is no need to expose the `__consumer_offsets` alias via the Admin API. 
+We may assume that Admin API users are aware of Redpanda internals and 
+we may operate using the `kafka_internal/group` Redpanda-specific topic name.
+
+
+## Corner cases dissected by example
+
+### Topic `kafka/__consumer_offsets` already exists in the cluster
+
+In this case we will just use whatever is there and ignore the alias.
+
+### Topic `kafka/__consumer_offsets` is created by the user
+
+In this case, Redpanda will create an underlying `kafka_internal/groups` topic with 
+a specified set of options provided by a caller. We may emit a WARN log message 
+that the topic being created is an internal topic. 
+
+## Detailed design - What needs to change to get there
+
+### Internal topics
+
+We will introduce a `bool is_internal_topic(const model::topic&)` function
+that will return true whenever a topic is internal. This will provide 
+compatibility with Kafka internal topic abstraction. Topics are marked as 
+internal in `MetadataResponse`. 
+
+### Topic aliases
+
+Add an `std::vector<topic_alias>` sequence container to the topic table where topic aliases will be 
+registered. We can tolerate the linear lookup const since the number of aliases 
+is very limited (currently 2).
+
+```c++
+struct  topic_alias {
+    model::topic_namespace alias;
+    model::topic_namespace source;
+};
+```
+
+`cluster::metadata_cache` will transparently handle all APIs for aliased topics.
+
+NOTE: `topic_alias` data structure is generic. However, for simplicity in the
+Kafka API handlers, we will handle aliases for topics marked as `is_internal`.
+
+### Translation layer
+
+We must add a data structure that will hold the data transformation factor related to the given 
+topic alias. Reference to that data structure will be provided by request via 
+`kafka::request_context`. Data transformation will be a simple class, 
+creating `model::record_batch_reader` instances that will apply a simple 
+transformation.
+
+
+## Drawbacks
+
+The main drawback of the proposed solution is that it must be special cased in 
+multiple places in the code. However, creating a generic internal topic alias 
+mechanism will allow us to more easily handle similar cases in the future,  
+such as support of a transactions topic.
+
+## Rationale and alternatives
+
+The main advantage of the approach presented here is the decoupling of the Redpanda implementation 
+from Kafka internals. An alternative approach would be to migrate all the data 
+from the `kafka_internal/group` topic to the new `kafka/__consumer_offsets`. 
+This, however, would bound the Redpanda implementation to the format of the data that 
```

Comment:
Migrating `kafka_internal/group` to `kafka/__consumer_offsets` still looks very tempting. The scope of this RFC is huge: `MetadataResponse`, `ListOffsets`, `DescribeConfigs`, `DescribeLogDirs`, `AlterConfig`, `IncrementalAlterConfig`, `CreateTopic`, `Fetch`, `Produce`.

While if we migrate the group topic we'll achieve the similar effect without all this work. Also it isn't obvious that it bounds us to Kafka format, we already have an offset translation layer which maps raft log to kafka log, we may do the same for `kafka/__consumer_offsets` **when and if** it diverges from our internal format.